### PR TITLE
[INLONG-9784][Audit] Optimize sending memory management when the audit-proxy config is null

### LINKED ISSUE #9784: [Improve][Audit] Optimize sending memory management
### Description

Optimize sending memory management of audit sdk

### InLong Component

InLong Audit

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by doleyzi**
### Prepare a Pull Request
- Fixes #9784
### Motivation

*Optimize sending memory management when the audit-proxy config is null.*

### Modifications

*Add memory release logic when the audit-proxy config is null.*

### CODE REVIEW

**Review by luchunliang — APPROVED** [review]

File: inlong-audit/audit-sdk/src/main/java/org/apache/inlong/audit/send/SenderGroup.java [hunk 1]
```diff
@@ -83,10 +83,12 @@ public SenderGroup(SenderManager senderManager) {
     public SenderResult send(ByteBuf dataBuf) {
         LinkedBlockingQueue<SenderChannel> channels = channelGroups.get(mIndex);
         SenderChannel channel = null;
+        boolean dataBufHasRelease = false;
```
Comment (justinwwhuang) [inline inlong-audit/audit-sdk/src/main/java/org/apache/inlong/audit/send/SenderGroup.java:86]: dataBufHasReleased?

**Review by dockerzhang — APPROVED** [review]

**Review by justinwwhuang — DISMISSED** [review]
LGTM

File: inlong-audit/audit-sdk/src/main/java/org/apache/inlong/audit/send/SenderGroup.java [hunk 2]
```diff
@@ -155,8 +160,12 @@ public SenderResult send(ByteBuf dataBuf) {
             return new SenderResult("127.0.0.1", 0, false);
         } finally {
             if (channel != null) {
+                channel.release();
                 channels.offer(channel);
             }
+            if (!dataBufReleased && dataBuf != null) {
```
Comment (aloyszhang) [inline inlong-audit/audit-sdk/src/main/java/org/apache/inlong/audit/send/SenderGroup.java:166]: Nullable check for `dataBuf` at the start place may be better.

**Review by aloyszhang — APPROVED** [review]
LGTM

**Review by justinwwhuang — APPROVED** [review]
LGTM

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**